### PR TITLE
FPM writer: do not overwrite files

### PR DESCRIPTION
**Pull request by stefanschreck**
Solves feature request https://github.com/SAP/open-ux-tools/issues/263.

All extension writers are enhanced so that they do not overwrite existing fragments, views, or controllers. 
Instead, the given file is taken but the program continues and refers the existing file in the manifest.
New tests have been added for this scenario.

For further information please check the feature request.

### CODE REVIEW

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Looks good. I have just two minor requests for improvements.
I didn't run the unit tests locally but the snapshot updates look correct and the CI test runs succeed, so I assume they are good.

File: packages/fe-fpm-writer/src/section/index.ts [hunk 1]
```diff
@@ -70,12 +70,11 @@ export function generateCustomSection(basePath: string, customSection: CustomSec
...
     // add event handler if requested
-    if (completeSection.eventHandler) {
-        fs.copyTpl(
-            join(root, 'common/EventHandler.js'),
-            join(completeSection.path, `${completeSection.name}.js`),
-            completeSection
-        );
+    const controllerPath = join(completeSection.path, `${completeSection.name}.js`);
+    if (!fs.exists(controllerPath)) {
+        if (completeSection.eventHandler) {
```
[the remark this answers is not in the document]
Comment (stefanschreck) [reply packages/fe-fpm-writer/src/section/index.ts:75]: right, I will change this.

File: packages/fe-fpm-writer/test/unit/column.test.ts [hunk 2]
```diff
@@ -80,7 +82,27 @@ describe('CustomAction', () => {
             expect(fs.read(expectedFragmentPath)).toMatchSnapshot();
             expect(fs.read(expectedFragmentPath.replace('.fragment.xml', '.js'))).toMatchSnapshot();
         });
-        test('generateCustomColumn 1.85, no handler, all properties', () => {
+        test('version 1.86, with existing handler, all properties', () => {
+            const controllerPath = 'my.test.App.ext.ExistingHandler.onCustomAction';
+            fs.write(controllerPath, 'dummyContent');
+            const testCustomColumn: CustomTableColumn = {
+                ...customColumn,
+                eventHandler: controllerPath,
```
[the remark this answers is not in the document]
Comment (stefanschreck) [reply packages/fe-fpm-writer/test/unit/column.test.ts]: @tobiasqueck done, please check again

File: packages/fe-fpm-writer/test/unit/section.test.ts [hunk 3]
```diff
@@ -136,7 +152,7 @@ describe('CustomSection', () => {
             expect(fs.read(fragmentPath)).toMatchSnapshot();
         });
 
-        const testVersions = [1.9, 1.85, 1.84,  1.86, 1.90];
+        const testVersions = [1.9, 1.85, 1.84, 1.86, 1.9];
```
Comment (tobiasqueck) [inline packages/fe-fpm-writer/test/unit/section.test.ts:155]: that looks wrong. Why `1.9` and not `1.90`, btw, now you have `1.9` twice in the array

File: packages/fe-fpm-writer/tsconfig.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-    "include": ["src"],
+    "include": ["src/**/*.ts"],
```
Comment (tobiasqueck) [inline packages/fe-fpm-writer/tsconfig.json:2]: Why this change?
We have `src` in every other project e.g. https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/tsconfig.json#L2

File: packages/fe-fpm-writer/test/unit/section.test.ts (hunk 3 above)
Comment (stefanschreck) [reply packages/fe-fpm-writer/test/unit/section.test.ts:155]: was a mistake. I'll undo.

File: packages/fe-fpm-writer/tsconfig.json (hunk 4 above)
Comment (stefanschreck) [reply packages/fe-fpm-writer/tsconfig.json:2]: I tried to get rid of my local eslint problems, and had tried several things. When reading the [documentation](https://www.typescriptlang.org/tsconfig#include), I thought it better to restrict to the ts files here. But as .ts, .tsx, and .d.ts are the default we can also neglect the suffix. Indeed, it is not the only tsconfig file at open-ux-tools which uses the suffix. Finally, it might better to work without suffix as I can also find a glob definition under https://github.com/SAP/open-ux-tools/blob/main/tsconfig.json.

Comment (stefanschreck) [reply packages/fe-fpm-writer/tsconfig.json]: @tobiasqueck all done

File: packages/fe-fpm-writer/test/unit/section.test.ts (hunk 3 above)
Comment (tobiasqueck) [reply packages/fe-fpm-writer/test/unit/section.test.ts:155]: @stefanschreck  You re-requested a review but I still see the issue. Did you forget to push something?

Comment (stefanschreck) [reply packages/fe-fpm-writer/test/unit/section.test.ts:155]: weird. the commit was pushed but without this correction. need a new commit.

Comment (stefanschreck) [reply packages/fe-fpm-writer/test/unit/section.test.ts]: @tobiasqueck the change gets condensed during the commit, as 1.9 and 1.90 are the same **numbers**. I can only choose a different number, e.g. 1.98.

**Review by tobiasqueck — APPROVED** [review]

**Comment by stefanschreck** [comment]
Thanks, Tobias.
@815are @Klaus-Keller additional reviewer needed...

**Review by 815are — APPROVED** [review]
